lib/mayaUsd/fileio/primUpdaterManager.h

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef PXRUSDMAYA_MAYAPRIMUPDATER_MANAGER_H
+#define PXRUSDMAYA_MAYAPRIMUPDATER_MANAGER_H
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/fileio/primUpdaterContext.h>
+#include <mayaUsd/listeners/proxyShapeNotice.h>
+#include <mayaUsd/utils/util.h>
+
+#include <pxr/base/tf/registryManager.h>
+#include <pxr/pxr.h>
+#include <pxr/usd/sdf/path.h>
+
+#include <maya/MCallbackIdArray.h>
+#include <maya/MDagPath.h>
+#include <maya/MFnDependencyNode.h>
+#include <maya/MObject.h>
+#include <ufe/path.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+class PrimUpdaterManager : public PXR_NS::TfWeakBase
+{
+public:
+    MAYAUSD_CORE_PUBLIC
+    ~PrimUpdaterManager();
+
+    MAYAUSD_CORE_PUBLIC
+    bool Push(const MFnDependencyNode& depNodeFn, const Ufe::Path& pulledPath);
+
+    MAYAUSD_CORE_PUBLIC
+    bool Pull(const Ufe::Path& path);
+
+    MAYAUSD_CORE_PUBLIC
+    bool PullClear(const Ufe::Path& path);
+
+    MAYAUSD_CORE_PUBLIC
+    bool CopyBetween(const Ufe::Path& srcPath, const Ufe::Path& dstPath);
+
+    /// \brief Returns the singleton prim updater manager
+    MAYAUSD_CORE_PUBLIC
+    static PrimUpdaterManager& GetInstance();
+
+private:
+    PrimUpdaterManager();
+
+    void onProxyContentChanged(const MayaUsdProxyStageObjectsChangedNotice& notice);
+
+    //! Ensure the Dag pull root exists.  This is the child of the Maya world
+    //! node under which all pulled nodes are created.
+    bool FindOrCreatePullRoot();
```

Comment:
We have some inconsistency in naming convention here. Upper case vs lower case method names.